src/ParkChecklist/ParkChecklist.js

```diff
@@ -0,0 +1,94 @@
+import page from "../resources/page.png";
+
+const ParkChecklist = ({ apiLink }) => {
+    const [parks, setParks] = useState([]);
+    const [checkedItems, setCheckedItems] = useState([]);
+    const [selectedState, setSelectedState] = useState('');
+    const [showResults, setShowResults] = useState(false);
+    const [currentPage, setCurrentPage] = useState(1);
+    const parksPerPage = 50;
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const response = await axios.get(`${apiLink}&limit=500`);
+                setParks(response.data.data);
+            } catch (error) {
+                console.error('Error fetching data:', error);
+            }
+        };
+
+        fetchData();
+    }, [apiLink, selectedState, showResults, currentPage]);
+
+    const handleCheckboxChange = (parkId) => {
+        setCheckedItems((prevCheckedItems) => {
+            if (prevCheckedItems.includes(parkId)) {
+                return prevCheckedItems.filter((id) => id !== parkId);
+            } else {
+                return [...prevCheckedItems, parkId];
+            }
+        });
+    };
+
+    const handleStateChange = (event) => {
+        setSelectedState(event.target.value);
+    };
+
+    const handleGoClick = () => {
+        setShowResults(true);
+    };
+
+    const handlePageChange = (newPage) => {
+        setCurrentPage(newPage);
+    };
+
+    const totalChecked = checkedItems.length;
+    const indexOfLastPark = currentPage * parksPerPage;
+    const indexOfFirstPark = indexOfLastPark - parksPerPage;
+    const currentParks = parks.slice(indexOfFirstPark, indexOfLastPark);
+
+
+    return (
+        <div className="park-checklist-container" style={{ backgroundImage: `url(${page})` }}>
+            <h1>National Park Service Service</h1>
+            <div className='below-header-box'>
+                <p className='welcome-message'>Welcome User!</p>
+                <div className='state-selector'>
+                    <select onChange={handleStateChange} value={selectedState}>
+                        <option value="">Select a State</option>
+                        <option value="AL">Alabama</option>
+                    </select>
+                    <button onClick={handleGoClick}>Go!</button>
+                </div>
+            </div>
+            <h2>{`${totalChecked}/${parks.length} parks visited`}</h2>
+            <h2>Parks Checklist</h2>
+            <ul className="checkbox-list">
+                {currentParks.map((park) => (
+                    <li key={park.id}>
+                        <input
+                            type="checkbox"
+                            id={park.id}
+                            checked={checkedItems.includes(park.id)}
+                            onChange={() => handleCheckboxChange(park.id)}
+                        />
+                        <label htmlFor={park.id}>{park.fullName}</label>
+                    </li>
+                ))}
```

Comment:
Will the state of whether a box has been checked be saved here, or will that info need to be passed to the BE server for saving into the DB?

Reply:
it should be passed to both actually. the `useState` tracks the boxes that have been ticked and updates based on that, and then it will need to be stored to the DB. the state just shows it visibly on this page